src/x86_64/lock.h

```diff
@@ -17,7 +17,8 @@ static inline void spin_lock(spinlock l) {
 
 static inline void spin_unlock(spinlock l) {
     compiler_barrier();
-    *(volatile u64 *)&l->w = 0;
+    volatile word *w = &l->w;
```

Comment:
I found the type aliasing error was being caused by stage2 getting compiled with these - but stage2 doesn't need spinlocks. Were you also getting those errors during stage2 build?

I suggest changing #ifdef SMP_ENABLE to #if defined(KERNEL) && defined(SMP_ENABLE).

Reply:
Indeed the error only occurs when building stage2. I reverted the change and used `#if defined(KERNEL) && defined(SMP_ENABLE)` instead.